lib/screens/home_screen.dart

```diff
@@ -61,7 +66,49 @@ class _HomeScreenState extends State<HomeScreen> {
   void _toggleTaskCompletion(int index, bool? value) async {
     setState(() {
       tasks[index].isCompleted = value ?? false;
+
+      if (tasks[index].isCompleted) {
+        if (tasks[index].hasDeadline) {
+          var days_diff =
+              tasks[index].deadline.difference(DateTime.now()).inDays;
+          kudos.score += days_diff;
+
+          String status = (days_diff > 0)
+              ? "$days_diff days before deadline"
+              : (days_diff == 0)
+                  ? "on time"
+                  : "${-days_diff} days after deadline";
+
+          String title = "Completed '${tasks[index].title}' $status";
```

Comment:
In general, we write all string greetings in a seperate file called constants.dart because it is easier for internationalization and changing stuff ahead, please do so instead of writing here. U can say something like completeOnTimeSuffix and define it there